### PR TITLE
Feature/issue 833 team position

### PR DESCRIPTION
**Pull request by BapBap7**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.BLL/MediatR/Team/Position/Create/CreatePositionHandler.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Team/Position/Create/CreatePositionHandler.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/MediatR/Team/Position/Delete/DeleteTeamPositionCommand.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Team/Position/Delete/DeleteTeamPositionCommand.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/MediatR/Team/Position/Delete/DeleteTeamPositionHandler.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Team/Position/Delete/DeleteTeamPositionHandler.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/MediatR/Team/Position/GetByTitle/GetByTitleTeamPositionQuery.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Team/Position/GetByTitle/GetByTitleTeamPositionQuery.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/MediatR/Team/Position/Update/UpdateTeamPositionCommand.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Team/Position/Update/UpdateTeamPositionCommand.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.BLL/MediatR/Team/Position/Update/UpdateTeamPositionHandler.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.BLL/MediatR/Team/Position/Update/UpdateTeamPositionHandler.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/TeamPositionsControllerTests.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/TeamPositionsControllerTests.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/AdditionalContent/Positions/ExtractCreateTestPosition.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/AdditionalContent/Positions/ExtractCreateTestPosition.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/AdditionalContent/Positions/ExtractUpdateTestPosition.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/AdditionalContent/Positions/ExtractUpdateTestPosition.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Client/Additional/TeamPositionsClient.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Client/Additional/TeamPositionsClient.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Extracter/AdditionalContent/TeamPositionsExtracter.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Extracter/AdditionalContent/TeamPositionsExtracter.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/CreatePositionTest.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/CreatePositionTest.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/DeletePositionTest.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/DeletePositionTest.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/GetAllWithTeamMembersTest.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/GetAllWithTeamMembersTest.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/GetByIdPositionTest.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/GetByIdPositionTest.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/GetByTitlePositionTest.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/GetByTitlePositionTest.cs:1]: Add new line at the end of file

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/UpdatePositionTest.cs
Comment (Lazy-Lenny) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Team/Position/UpdatePositionTest.cs:1]: Add new line at the end of file

**Review by Lazy-Lenny — APPROVED** [review]

**Review by Spoilsport0 — APPROVED** [review]

**Review by Dobriyr — APPROVED** [review]
good job

File: Streetcode/Streetcode.BLL/MediatR/Team/Position/Delete/DeleteTeamPositionHandler.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    public DeleteTeamPositionHandler(IRepositoryWrapper repository, ILoggerService logger)
+    {
+        _repository = repository;
+        _logger = logger;
+    }
+
+    public async Task<Result<int>> Handle(DeleteTeamPositionCommand request, CancellationToken cancellationToken)
+    {
+        var positionToDelete =
```
Comment (Dobriyr) [inline Streetcode/Streetcode.BLL/MediatR/Team/Position/Delete/DeleteTeamPositionHandler.cs:21]: You can reduce and optimize this code by making just one database query. This option is valid, but it is better to use BulkDelete
like this 
_dbContext.Positions.Where(x => x.Id == id).ExecuteDelete();

**Pull request merged**